### PR TITLE
Added theta parameter to Gaussian2DKernel

### LINKED ISSUE #3605: Add theta parameter to GaussianKernel2D
GaussianKernel2D currently does not take a position angle (as far as I can tell), but it would be good to implement this:

http://docs.astropy.org/en/stable/api/astropy.convolution.Gaussian2DKernel.html#astropy.convolution.Gaussian2DKernel

@adonath - any reason why it should not be included?

### PR DESCRIPTION
**Pull request by pllim**
Fix #3605 

Close #4687

### CODE REVIEW

File: astropy/convolution/kernels.py [hunk 1]
```diff
@@ -97,8 +97,12 @@ class Gaussian2DKernel(Kernel2D):
 
     Parameters
     ----------
-    stddev : number
-        Standard deviation of the Gaussian kernel.
+    x_stddev : float
+        Standard deviation of the Gaussian in x.
```
Comment (MSeifert04) [inline astropy/convolution/kernels.py]: Could you add a `..versionadded: 3.0 Replaces the deprecated stddev argument`?

File: astropy/convolution/kernels.py [hunk 2]
```diff
@@ -97,8 +97,12 @@ class Gaussian2DKernel(Kernel2D):
...
+    y_stddev : float
+        Standard deviation of the Gaussian in y.
+    theta : float
+        Rotation angle in radians. Note: increases clockwise.
```
Comment (MSeifert04) [inline astropy/convolution/kernels.py]: That sounds odd with "increases clockwise"...

Comment (pllim) [reply astropy/convolution/kernels.py]: Ops, should that be counter clockwise like Gaussian2D model?

Comment (MSeifert04) [reply astropy/convolution/kernels.py]: Probably, but the parameter description there is also a bit "better": "The rotation angle increases counterclockwise."

Comment (larrybradley) [reply astropy/convolution/kernels.py]: @pllim The rotation angle increases *counterclockwise*.

Comment (pllim) [reply astropy/convolution/kernels.py]: Yes, my bad. I didn't proof read the doc. I'll fix that in my next commit.

**Comment by pllim** [comment]
The only thing I am not sure about is the math in `__init__` (which I copied over from original PR) but since @adonath approved that, should be okay? I still need an approval from someone with write access though.

**Comment by pllim** [comment]
Thanks!

**Pull request merged**